marshal.go

```diff
@@ -105,14 +105,19 @@ func parsePort(s string) int {
 // the raw JSON for later parsing.
 func (i *InstanceMetadata) UnmarshalJSON(b []byte) error {
 	i.Raw = b
-	// TODO(cq) could actually parse Raw here, and in a parallel UnmarshalXML as well.
+	// TODO(cq) could actually parse Raw in a parallel UnmarshalXML as well.
+	var m map[string]interface{}
+	if err := json.Unmarshal(b, &m); err != nil {
+    		return err
+	}
+	i.Parsed = m
 	return nil
 }
 
 // MarshalJSON is a custom JSON marshaler for InstanceMetadata.
 func (i *InstanceMetadata) MarshalJSON() ([]byte, error) {
-	if i.parsed != nil {
-		return json.Marshal(i.parsed)
+	if i.Parsed != nil {
```

Comment:
Should that be checking the entry count instead? That is, if there's a map instance present, but it's empty, should that still be marshaled?
```go
if len(i.Parsed) > 0 {
```